### PR TITLE
Remove outdated trainer argument videos

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Removes references to videos that use outdated APIs or examples

Feel free to push to this PR making edits to descriptions, if you think something should be written from the removed videos.

cc @borda

### CODE REVIEW

File: docs/source-pytorch/common/gradient_accumulation.rst [hunk 1]
```diff
@@ -9,8 +9,6 @@ effective batch size is increased but there is no memory overhead.
     step, the effective batch size on each device will remain ``N*K`` but right before the ``optimizer.step()``, the gradient sync will make the effective
     batch size as ``P*N*K``. For DP, since the batch is split across devices, the final effective batch size will be ``N*K``.
 
-.. seealso:: :class:`~lightning.pytorch.trainer.trainer.Trainer`
```
Comment (carmocca) [inline docs/source-pytorch/common/gradient_accumulation.rst:12]: Not sure how this link is helpful. This just links to the API reference

File: docs/source-pytorch/common/trainer.rst [hunk 2]
```diff
@@ -239,14 +239,6 @@ Example::
 accumulate_grad_batches
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-.. raw:: html
```
Comment (carmocca) [inline docs/source-pytorch/common/trainer.rst:242]: Outdated after https://github.com/Lightning-AI/lightning/pull/16729

File: docs/source-pytorch/common/trainer.rst [hunk 3]
```diff
@@ -316,23 +308,18 @@ Example::
 callbacks
 ^^^^^^^^^
 
-.. raw:: html
-
-    <video width="50%" max-width="400px" controls
```
Comment (carmocca) [inline docs/source-pytorch/common/trainer.rst:321]: Mentions bolts and shows several legacy hooks

File: docs/source-pytorch/common/trainer.rst [hunk 4]
```diff
@@ -452,14 +439,6 @@ Number of devices to train on (``int``), which devices to train on (``list`` or
 enable_checkpointing
 ^^^^^^^^^^^^^^^^^^^^
 
-.. raw:: html
```
Comment (carmocca) [inline docs/source-pytorch/common/trainer.rst:455]: Empty

File: docs/source-pytorch/common/trainer.rst [hunk 5]
```diff
@@ -663,14 +640,6 @@ See Also:
 logger
 ^^^^^^
 
-.. raw:: html
```
Comment (carmocca) [inline docs/source-pytorch/common/trainer.rst:666]: Empty

File: docs/source-pytorch/common/trainer.rst [hunk 6]
```diff
@@ -869,14 +838,6 @@ Useful for quickly debugging or trying to overfit on purpose.
 plugins
 ^^^^^^^
 
-.. raw:: html
```
Comment (carmocca) [inline docs/source-pytorch/common/trainer.rst:872]: Empty

File: docs/source-pytorch/common/trainer.rst [hunk 7]
```diff
@@ -907,14 +868,6 @@ To define your own behavior, subclass the relevant class and pass it in. Here's
 precision
 ^^^^^^^^^
 
-.. raw:: html
```
Comment (carmocca) [inline docs/source-pytorch/common/trainer.rst:910]: Mentions various outdated facts.

File: docs/source-pytorch/extensions/callbacks.rst [hunk 8]
```diff
@@ -7,24 +7,14 @@
 Callback
 ########
 
-.. raw:: html
```
Comment (carmocca) [inline docs/source-pytorch/extensions/callbacks.rst:10]: Same as trainer arg comment

File: docs/source-pytorch/extensions/callbacks.rst [hunk 9]
```diff
@@ -7,24 +7,14 @@
...
+It de-couples functionality that does not need to be in the :doc:`lightning module <../common/lightning_module>` and can be shared across projects.
 
 Lightning has a callback system to execute them when needed. Callbacks should capture NON-ESSENTIAL
 logic that is NOT required for your :doc:`lightning module <../common/lightning_module>` to run.
 
-Here's the flow of how the callback hooks are executed:
-
-.. raw:: html
-
-    <video width="100%" max-width="400px" controls autoplay muted playsinline src="https://pl-bolts-doc-images.s3.us-east-2.amazonaws.com/pl_docs/pt_callbacks_mov.m4v"></video>
```
Comment (carmocca) [inline docs/source-pytorch/extensions/callbacks.rst:27]: Shows several outdated hooks

File: docs/source-pytorch/extensions/callbacks.rst [hunk 10]
```diff
@@ -54,19 +44,8 @@ Example:
...
 :doc:`lightning module <../common/lightning_module>` research code.
 
------------
-
-********
-Examples
-********
 You can do pretty much anything with callbacks.
 
-- `Add a MLP to fine-tune self-supervised networks <https://lightning-bolts.readthedocs.io/en/latest/callbacks/self_supervised.html#sslonlineevaluator>`_.
```
Comment (carmocca) [inline docs/source-pytorch/extensions/callbacks.rst:64]: Bolts is outdated

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**